src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -583,15 +592,20 @@ private void retryableExecuteAgent(
 
         mlClient.execute(FunctionName.AGENT, agentMLInput, ActionListener.wrap(response -> {
             MLOutput mlOutput = (MLOutput) response.getOutput();
-            String result = null;
+            String dslQuery = null;
+            String agentStepsSummary = null;
+
+            String memoryId = null;
             if (type == MLAgentType.FLOW) {
-                result = extractFlowAgentResult(mlOutput);
+                dslQuery = extractFlowAgentResult(mlOutput);
```

Comment:
We might want to throw exception when memory id is provided with flow type? Otherwise, user might not know if their request is utilizing memory or not if it get ignored silently?

Reply:
Good catch. Will update and add a test